### PR TITLE
Documentation for Analytics and Consent Management Implementation

### PR DESCRIPTION
**Pull request by mark-pkgflow**
## Overview

This PR introduces high-level documentation, describing our implementation strategy for Google Analytics 4 (GA4), Google Tag Manager (GTM), the use of cookies, and the user consent banner on our website.

## Purpose

The purpose of this documentation is to:

- Provide clarity on how we implement and utilise GA4 and GTM.
- Outline our use of cookies in compliance with privacy regulations.
- Explain the functionality and necessity of the user consent banner.
- Detail the data privacy and user control measures we have in place.

## Impact

- No impact on the existing codebase or functionality.
- Provides a reference for team members and stakeholders on our consent management practices.
- Ensures transparency in our analytics and data collection methods.

## How to Review

- Please read through the document to verify the accuracy of the described processes and configurations.
- Ensure that the document is understandable and clearly communicates our practices to both technical and non-technical stakeholders.

### CODE REVIEW

**Comment by mark-pkgflow** [comment]
[Hook up Google consent mode to the cookie banner and cookie form](https://trello.com/c/nsLeox3l/338-hook-up-google-consent-mode-to-the-cookie-banner-and-cookie-form)

File: docs/analytics-overview.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+- Configuration of event tracking for user interactions without altering the source code.
+- Managing and testing marketing tags and related code snippets.
+
+## Cookies
+
+Cookies are small data files stored on the user's device, which we use to:
+
+- Remember user preferences and sessions.
+- Collect information that assists with personalizing the user experience.
+- Track user activity for analytics and advertising purposes in compliance with our privacy policy.
```
Comment (AbigailMcP) [inline docs/analytics-overview.md:32]: Making it more applicable to our service:

```suggestion
- Remember user preference for analytics tracking.
- Track user activity for analytics and advertising purposes in compliance with our privacy policy.
```

File: docs/analytics-overview.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+We respect user privacy and comply with legal requirements for data protection. As such, we implement a consent banner that:
+
+- Informs users about the use of cookies and tracking technologies on our website.
+- Allows users to give, or deny their consent for data collection.
+- Ensures that no tracking cookies are placed without explicit user consent, complying with GDPR, CCPA, and other data
+protection laws.
+
+## Consent Management and Data Collection Process
+
+When a user visits our website, the following occurs:
```
Comment (AbigailMcP) [inline docs/analytics-overview.md:45]: We could also say somewhere in this section that before indicating their consent preference, `consentStatus` is set to `false`.

File: docs/analytics-overview.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+2. If the user consents, `consentStatus` is set to `true` in the data layer and GA4 tracking is fully enabled.
+3. If the user does not consent, `consentStatus` is set to `false` and GA4 tracking is limited to anonymized, aggregate data.
+4. GTM uses the `consentStatus` variable to control the firing of tags based on the user's consent.
+5. The website will respect the user's choice, and their consent preference will be stored in cookies for future visits.
+
+## Data Privacy and User Control
+
+Our users have control over their data. They can:
+
+- Change their consent preferences at any time via the consent banner or website settings.
```
Comment (AbigailMcP) [inline docs/analytics-overview.md:57]:
```suggestion
- Change their consent preferences at any time via the consent banner or cookies policy page.
```

**Review by AbigailMcP — APPROVED** [review]

**Comment by mark-pkgflow** [comment]
@claire-hughez to see it in rendered markdown you can check [here](https://github.com/DFE-Digital/improve-school-workload-and-wellbeing/blob/8ce5694b13608767258f56436203321d5202175a/docs/analytics-overview.md).

File: docs/analytics-overview.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+This document provides a high-level overview of how the Improve workload and wellbeing for school staff service utilises
+Google Analytics 4 (GA4), Google Tag Manager (GTM), cookies, and a consent banner to track user interactions and comply
+with privacy regulations.
+
+## Google Analytics 4 (GA4)
+
+GA4 is the latest iteration of Google's analytics service, providing insights into website traffic, user behavior, and
+conversion metrics. We use GA4 to:
+
+- Monitor key metrics and performance indicators of our digital presence.
```
Comment (claire-hughez) [inline docs/analytics-overview.md:14]:
```suggestion
- Monitor key metrics and performance indicators of our service.
```

File: docs/analytics-overview.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+- Understand user paths and interactions with our website.
```
Comment (claire-hughez) [inline docs/analytics-overview.md:15]:
```suggestion
- Understand user behaviour and interactions with our website.
```

File: docs/analytics-overview.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- Enhance marketing strategies and user experience based on data-driven insights.
```
Comment (claire-hughez) [inline docs/analytics-overview.md:16]:
```suggestion
- Improve the service based on data-driven insights.
```

**Review by claire-hughez — COMMENTED** [review]
Looks great Mark, just a few tweaks for why we're using it.

Do we have any more clarity on how the data is impacted/different from when users reject cookies too?

**Comment by mark-pkgflow** [comment]
> Do we have any more clarity on how the data is impacted/different from when users reject cookies too?

Good question @claire-hughez 

I have added a section for this:

## Data Collection with and without Consent Mode

**When consent is given:** Google Analytics functions as normal, collecting full user data including cookies for user
identification, pageview hits, event hits, ecommerce hits, and so on.

**When consent is not given or is withdrawn:** GA4 enters a restricted data collection mode. In this state, it does not
use cookies for user identification, which means it does not collect any data that is considered personally identifiable
information (PII). Instead, it uses a session-based model for limited data collection, which can include:

- Aggregated data about traffic and general site interaction without user identifiers.
- Conversion tracking (with limitations, as no user-level data is recorded).
- Basic interaction hits (such as page views or non-identifying event hits) with anonymized IP addresses.

**Pull request merged**